tests/components/imeon_inverter/test_config_flow.py

```diff
@@ -0,0 +1,157 @@
+"""Test the Imeon Inverter config flow."""
+
+from collections.abc import Generator
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.imeon_inverter.const import DOMAIN
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .conftest import TEST_SERIAL, TEST_USER_INPUT
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("mock_async_setup_entry")
+
+
+async def test_form(
+    hass: HomeAssistant,
+    mock_async_setup_entry: Generator[AsyncMock],
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], TEST_USER_INPUT
+    )
+    await hass.async_block_till_done()
+
+    assert result2["type"] is FlowResultType.CREATE_ENTRY
+    assert result2["title"] == TEST_USER_INPUT[CONF_ADDRESS]
+    assert result2["data"] == TEST_USER_INPUT
+    assert mock_async_setup_entry.call_count == 1
```

Comment:
```suggestion
    result = await hass.config_entries.flow.async_configure(
        result["flow_id"], TEST_USER_INPUT
    )
    await hass.async_block_till_done()

    assert result["type"] is FlowResultType.CREATE_ENTRY
    assert result["title"] == TEST_USER_INPUT[CONF_ADDRESS]
    assert result["data"] == TEST_USER_INPUT
    assert mock_async_setup_entry.call_count == 1
```

we can actually override the result all the time as we no longer need it (for the rest as well)